setup/index.ts

```diff
@@ -1,11 +1,31 @@
-import { askSessionToken } from './prompt';
-import { get as getEnvValue, set as setEnvValue } from './env';
+import {
+  /* askSessionToken, */
+
+  askIOBackendHost,
+  askIOBackendBasePath
+} from "./prompt";
+import { get as getEnvValue, set as setEnvValue, EnvKey } from "../lib/env";
+import { TaskEither } from "fp-ts/lib/TaskEither";
+import { task } from "fp-ts/lib/Task";
+
+const promptIfNot = (
+  key: EnvKey,
+  promptFn: () => TaskEither<Error, string>
+) => {
+  const setter = () =>
+    promptFn().fold<string>(
+      error => { throw error },
+      value => {
+        setEnvValue(key, value);
+        return value;
+      }
+    );
+  return getEnvValue(key).fold(setter(), task.of);
+};
 
 const testSuiteSetup = async () => {
-    if(getEnvValue('SPID_SESSION_TOKEN')) return;
-    
-    const sessionToken = await askSessionToken();
-    setEnvValue('SPID_SESSION_TOKEN', sessionToken);
+  await promptIfNot("IO_BACKEND_HOST", askIOBackendHost).run();
```

Comment:
try to chain calls and call run() only once at the end. use sequence(T/S) methods and or traverse() to deal with arrays of taskeithers.

Reply:
see https://github.com/gcanti/fp-ts/issues/1140#issuecomment-589217217

Reply:
I'm trying applying this in many ways, but I cannot get it right. Sure I can group each task and run them at once, but I cannot have them to wait for the previous to be completed. This

```
  const backendHostTask = ask("IO_BACKEND_HOST", askIOBackendHost);
  const backendBasepathTask = ask("IO_BACKEND_BASEPATH", askIOBackendBasePath);

  const seq = array.sequence(taskEither);
  return seq([backendHostTask, backendBasepathTask])
    .fold(
      () => {
        throw requiredParamError();
      },
      value => value
    )
    .map(([IO_BACKEND_HOST, IO_BACKEND_BASEPATH]) => {
      setAllEnvValues({
        IO_BACKEND_HOST,
        IO_BACKEND_BASEPATH
      });
    })
    .run();
```

This is basically an implementation of `Promise.all`, where each task is run in parallel. How to have them run one after each other?

Reply:
https://grossbart.github.io/fp-ts-recipes/#/async?a=run-a-list-of-tasks-in-sequence

Reply:
thanks